src/source/Signaling/StateMachine.c

```diff
@@ -649,6 +708,13 @@ STATUS executeDisconnectedSignalingState(UINT64 customData, UINT64 time)
 
     CHK(pSignalingClient != NULL, STATUS_NULL_ARG);
 
+    // Notify of the state change
+    if (pSignalingClient->signalingClientCallbacks.stateChangeFn != NULL) {
```

Comment:
Can we make `stateChangeFn` mandatory?

I am afraid that in the future a NULL check may be missed and could cause an issue.

Reply:
Not sure if we need to make this a mandatory. What we do in KVS Producer SDK is we provide a default dummy implementation if the customer doesn't specify the functionality. We can add it later - not in this PR